### PR TITLE
Fix Kip's clock stuff in his branch of my branch.

### PR DESCRIPTION
**Pull request by CMurtagh-LGTM**
Branchception

### CODE REVIEW

File: module/platform/webots/Webots/src/Webots.cpp [hunk 1]
```diff
@@ -108,6 +108,52 @@ namespace module::platform::webots {
...
+            // We need to calculate the servo velocity to add to the velocity message
+            // (method stolen from HardwareIO.cpp)
+            // velocity = distance / time
+            const float distance =
+                utility::math::angle::difference(target.position, getDarwinServo(target.id, sensors).present_position);
+
+            NUClear::clock::duration time = target.time - NUClear::clock::now();
+            float velocity;
+            if (time.count() > 0) {
+                velocity = distance / static_cast<float>(time.count()) * static_cast<float>(NUClear::clock::period::num)
```
Comment (KipHamiltons) [inline module/platform/webots/Webots/src/Webots.cpp:131]: Probably worth making a PR to change `HardwareIO.cpp` too

File: module/platform/webots/Webots/src/Webots.cpp [hunk 2]
```diff
@@ -280,67 +272,71 @@ namespace module::platform::webots {
...
+                    ActuatorRequests to_send_now = make_acutator_request(commands, sensors);
+
+                    // ***************** Sending ********************************
+                    data = NUClear::util::serialise::Serialise<ActuatorRequests>::serialise(to_send_now);
+                    // Size of the message, in network endian
+                    Nn = htonl(data.size());
+                    // Send the message size first
+                    if (send(fd, &Nn, sizeof(Nn), 0) == sizeof(Nn)) {
+                        log<NUClear::ERROR>(
+                            fmt::format("Error in sending ActuatorRequests' message size,  {}", strerror(errno)));
```
Comment (KipHamiltons) [inline module/platform/webots/Webots/src/Webots.cpp:333]: `errno` isn't what you want it to be here

File: module/platform/webots/Webots/src/Webots.cpp [hunk 3]
```diff
@@ -280,67 +272,71 @@ namespace module::platform::webots {
...
+                    }
+                    // then send the data
+                    if (send(fd, data.data(), Nn, 0) == Nn) {
+                        log<NUClear::ERROR>(fmt::format("Error sending ActuatorRequests message, {}", strerror(errno)));
```
Comment (KipHamiltons) [inline module/platform/webots/Webots/src/Webots.cpp:337]: Same here

File: module/platform/webots/Webots/src/Webots.cpp [hunk 4]
```diff
@@ -349,9 +345,11 @@ namespace module::platform::webots {
 
         shutdown_handle = on<Shutdown>().then([this] {
             // Disconnect the fd gracefully
-            shutdown(this->fd, SHUT_RDWR);
-            close(this->fd);
-            this->fd = -1;
+            if (fd != -1) {
+                shutdown(fd, SHUT_RDWR);
+                close(fd);
+                fd = -1;
```
Comment (KipHamiltons) [inline module/platform/webots/Webots/src/Webots.cpp:351]: no idea why I needed `this` lol. Glad you could get rid of it

File: module/platform/webots/Webots/src/Webots.cpp [hunk 5]
```diff
@@ -108,6 +108,52 @@ namespace module::platform::webots {
         throw std::runtime_error("Unable to translate unknown NUgus.proto sensor name: " + name);
     }
 
+    ActuatorRequests make_acutator_request(const ServoTargets& commands, const DarwinSensors& sensors) {
```
Comment (KipHamiltons) [inline module/platform/webots/Webots/src/Webots.cpp:111]: Should we make this return a ref?

File: module/platform/webots/Webots/src/Webots.hpp [hunk 6]
```diff
@@ -30,9 +30,6 @@ namespace module::platform::webots {
 
     class Webots : public NUClear::Reactor {
     private:
-        /// @brief A single message that stores all the data that webots needs from us
-        message::platform::webots::ActuatorRequests to_send_now;
```
Comment (KipHamiltons) [inline module/platform/webots/Webots/src/Webots.hpp:34]: Also can remove `send_loop`

File: module/platform/webots/Webots/src/Webots.cpp (hunk 2 above)
Comment (CMurtagh-LGTM) [reply module/platform/webots/Webots/src/Webots.cpp:333]: :question:

Comment (CMurtagh-LGTM) [reply module/platform/webots/Webots/src/Webots.cpp:333]: https://man.archlinux.org/man/send.2.en#RETURN_VALUE says otherwise.

Comment (KipHamiltons) [reply module/platform/webots/Webots/src/Webots.cpp:333]: Oh shit. I forgot that shit was a meme global. mb

File: module/platform/webots/Webots/src/Webots.cpp (hunk 5 above)
Comment (CMurtagh-LGTM) [reply module/platform/webots/Webots/src/Webots.cpp:111]: But the Actuator request that we make is on the stack, so if we return a ref it will be garbage.

Comment (KipHamiltons) [reply module/platform/webots/Webots/src/Webots.cpp:111]: wouldn't it extend the lifetime???

**Review by KipHamiltons — APPROVED** [review]

**Pull request merged**